subworkflows/nf-core/gatk_tumor_normal_somatic_variant_calling/main.nf

```diff
@@ -0,0 +1,111 @@
+include { GATK4_FILTERMUTECTCALLS         as FILTERMUTECTCALLS }         from '../../../modules/gatk4/filtermutectcalls/main'         addParams( options: params.filtercalls_options )
+
+workflow GATK_TUMOR_NORMAL_SOMATIC_VARIANT_CALLING {
+    take:
+    ch_mutect2_in             // channel: [ val(meta), [ input ], [ input_index ], [which_norm] ]
+    fasta                     // channel: /path/to/reference/fasta
+    fai                       // channel: /path/to/reference/fasta/index
+    dict                      // channel: /path/to/reference/fasta/dictionary
+    germline_resource         // channel: /path/to/germline/resource
+    germline_resource_tbi     // channel: /path/to/germline/index
+    panel_of_normals          // channel: /path/to/panel/of/normals
+    panel_of_normals_tbi      // channel: /path/to/panel/of/normals/index
+    interval_file             // channel: /path/to/interval/file
+
+
+    main:
+    ch_versions = Channel.empty()
+    input = channel.from(ch_mutect2_in)
+
+    //
+    //Perform variant calling using mutect2 module in tumor single mode.
+    //
+    MUTECT2 ( input, false, false, false, [], fasta, fai, dict, germline_resource, germline_resource_tbi, panel_of_normals, panel_of_normals_tbi )
+    ch_versions = ch_versions.mix(MUTECT2.out.versions)
+
+    //
+    //Generate artifactpriors using learnreadorientationmodel on the f1r2 output of mutect2.
+    //
+    ch_learnread_in = MUTECT2.out.f1r2.collect()
+    LEARNREADORIENTATIONMODEL (ch_learnread_in)
+    ch_versions = ch_versions.mix(LEARNREADORIENTATIONMODEL.out.versions)
+
+    //
+    //Generate pileup summary tables using getepileupsummaries. tumor sample should always be passed in as the first input and input list entries of ch_mutect2_in,
+    //to ensure correct file order for calculatecontamination.
+    //
+    pileup_tumor_input = channel.from(ch_mutect2_in)
+    pileup_tumor_input = pileup_tumor_input.map {
+        meta, input_file, input_index, which_norm ->
+        [meta, input_file[0], input_index[0]]
+    }
+
+    pileup_normal_input = channel.from(ch_mutect2_in)
+    pileup_normal_input = pileup_normal_input.map {
+        meta, input_file, input_index, which_norm ->
+        [meta, input_file[1], input_index[1]]
+    }
+    GETPILEUPSUMMARIES_TUMOR ( pileup_tumor_input, germline_resource, germline_resource_tbi, interval_file )
+    GETPILEUPSUMMARIES_NORMAL ( pileup_normal_input, germline_resource, germline_resource_tbi, interval_file )
+    ch_versions = ch_versions.mix(GETPILEUPSUMMARIES_NORMAL.out.versions)
+
+    //
+    //Contamination and segmentation tables created using calculatecontamination on the pileup summary table.
+    //
+    ch_pileup     = GETPILEUPSUMMARIES_TUMOR.out.table.collect()
+    ch_pileup2    = GETPILEUPSUMMARIES_NORMAL.out.table.collect()
```

Comment:
instead of `_pileup` and `_pileup2` I'd go for `_pileup_tumor` and `_pileup_normal`